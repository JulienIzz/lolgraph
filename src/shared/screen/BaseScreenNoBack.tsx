import styled from "@emotion/native";
import { Header } from "./Header";

interface ScreenProps {
  children: React.ReactNode;
  title: string;
}

export const BaseScreenNoBack = ({ children, title }: ScreenProps) => {
  return (
    <Container>
      <Header title={title} />
      {children}
    </Container>
  );
};

const Container = styled.View({
  flex: 1,
});
