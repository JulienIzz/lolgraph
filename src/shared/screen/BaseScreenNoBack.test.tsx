import { render, screen } from "@testing-library/react-native";
import { SafeAreaProvider } from "react-native-safe-area-context";
import { BaseScreenNoBack } from "./BaseScreenNoBack";
import { View } from "react-native";

jest.mock(
  "react-native-safe-area-context",
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  () => require("react-native-safe-area-context/jest/mock").default,
);

test("home screen with title is correctly rendered", async () => {
  const expectedHeaderTextForTest = "ScreenTitle";

  render(
    <SafeAreaProvider>
      <BaseScreenNoBack title={expectedHeaderTextForTest}>
        <View />
      </BaseScreenNoBack>
    </SafeAreaProvider>,
  );

  const HeaderOutput = screen.getByText(expectedHeaderTextForTest);

  expect(HeaderOutput).toBeTruthy();
  expect(screen.toJSON()).toMatchSnapshot();
});
