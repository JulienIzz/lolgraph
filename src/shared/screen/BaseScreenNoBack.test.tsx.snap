// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home screen with title is correctly rendered 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    insets={
      {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      }
    }
    style={
      [
        {
          "backgroundColor": "#00B4D8",
          "height": 60,
          "paddingTop": 0,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 20,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        ScreenTitle
      </Text>
    </View>
  </View>
  <View />
</View>
`;
